Re-diffed for 2.6.14.7 by Kerin Millar <kerframil@gmail.com>:

From axboe@kernel.dk Mon Jan 30 05:47:34 2006
Date: Sun, 29 Jan 2006 20:57:33 +0100
From: Jens Axboe <axboe@suse.de>
To: James Bottomley <James.Bottomley@SteelEye.com>
Cc: Pasi =?iso-8859-1?Q?K=E4rkk=E4inen?= <pasik@iki.fi>,
	Nix <nix@esperi.org.uk>, Ariel <askernel2615@dsgml.com>,
	Jamie Heilman <jamie@audible.transient.net>,
	Chase Venters <chase.venters@clientec.com>,
	Arjan van de Ven <arjan@infradead.org>,
	linux-ide@vger.kernel.org, linux-kernel@vger.kernel.org,
	linux-scsi@vger.kernel.org
Subject: SCSI: turn off ordered flush barriers
Message-ID: <20060129195733.GH13831@suse.de>
Content-Disposition: inline

Turn off ordered flush barriers for SCSI driver, since the SCSI barrier
code has a command leak.

Signed-off-by: Jens Axboe <axboe@suse.de>


--- linux-2.6.14.7.orig/drivers/scsi/scsi_lib.c	2006-02-06 19:43:10.000000000 -0500
+++ linux-2.6.14.7/drivers/scsi/scsi_lib.c	2006-02-06 19:47:51.000000000 -0500
@@ -1579,11 +1579,6 @@ struct request_queue *scsi_alloc_queue(s
 	 */
 	if (shost->ordered_tag)
 		blk_queue_ordered(q, QUEUE_ORDERED_TAG);
-	else if (shost->ordered_flush) {
-		blk_queue_ordered(q, QUEUE_ORDERED_FLUSH);
-		q->prepare_flush_fn = scsi_prepare_flush_fn;
-		q->end_flush_fn = scsi_end_flush_fn;
-	}
 
 	if (!shost->use_clustering)
 		clear_bit(QUEUE_FLAG_CLUSTER, &q->queue_flags);
